src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -195,7 +197,7 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
                 var exceptions = new List<Exception>();
 
                 // Execute the hosting startup assemblies
-                foreach (var assemblyName in _options.HostingStartupAssemblies)
+                foreach (var assemblyName in _options.HostingStartupAssemblies.Distinct(StringComparer.OrdinalIgnoreCase))
```

Comment:
Let's not do that. You can't turn it off. Let's not do more things you can't turn off.